src/tree-node/node-label.js

```diff
@@ -51,7 +53,7 @@ class NodeLabel extends PureComponent {
     return (
       <label title={title || label} htmlFor={id}>
         <Checkbox
-          name={id}
+          name={`${clientId}[]`}
```

Comment:
Did you mean to keep the square brackets here?

Reply:
Yes, generating dropdownid[] as a name means all checkboxes gets handled as an array om form submits. At least for most server side receivers such as .NET MVC.

Reply:
Se https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/checkbox#Using_checkbox_inputs

Not sure how form submits are handled by the component even since they probably aren’t even rendered on the client on form submit? But either way that should be the way to handle grouping of releated checkboxes.

Reply:
Are you using it as such? I never intended the checkboxes to constitute the form values - mostly using the events to dispatch redux actions has been my primary use case. Interesting idea though...

Reply:
I know about the array notation, just that I never put much thought into making it act as a form control. I've used it with redux-form mostly where the flow I've used is `node change > onChange > dispatch (redux-form manages this) > submit (redux-form handles the form values`. I've also used it with just Redux but the flow is mostly the same - `onChange > dispatch > use store to submit values`.

Ok, so that aside, I think this can be merged (unless you're planning on making any more changes?).

Reply:
Oh, remembering the array notation and since you mentioned MVC - in order for serialization/deserialization to work properly, we'd need to render it without array notation in case of a single checkbox (with array notation it'll always be deserialized as an array and would thus fail if the receiving prop is not an array).

Also, a plain form submit may not be desirable since one may not want to submit all checkboxes all the time - e.g, with hierarchial mode, one may want to submit only the parent node if all children are selected.

If you need this feature, then let's open up a separate thread for this. Otherwise, let's stick to just ids for now. I want to start merging the PRs :)

Reply:
Actually it was based of your idea: https://github.com/dowjones/react-dropdown-tree-select/pull/220#discussion_r268414543

Do you want the unique node id back or the clientid without brackets? Current dev has a unique name per checkbox.

If we wanted the form submit to work we could just introduce a hidden field per tag with this suggested bracket naming and it should work with a regular form submit. Not something we use currently though. So this was all based on you suggestion.

Reply:
The old naming is restored, a.k.a no change on name for checkboxes

Reply:
So up for merge.

Reply:
> Actually it was based of your idea:

You missed the "I'm wondering..." part but I hear ya. My bad!

> Current dev has a unique name per checkbox.

Only works for single instances as you can see with #212.

> Do you want the unique node id back or the clientid without brackets?

Both actually! To fix #212, we need to make the checkbox id unique across instances. So I guess something like `${clientId}-${id}` should do the trick?

Reply:
The id is unique. The name is unique. It will work. No change is needed. The name is related to form submit only. Which web currently don’t support native I guess.